packages/dds/merge-tree/src/partialLengths.ts

```diff
@@ -362,15 +429,47 @@ export class PartialSequenceLengths {
                     ? PartialSequenceLengths.getOverlapClients(removeClientOverlap, segmentLen)
                     : undefined,
             };
-            if (indexFirstGTE < seqPartialsLen) {
-                // Shift entries with greater sequence numbers
-                // TODO: investigate performance improvement using BST
-                for (let k = seqPartialsLen; k > indexFirstGTE; k--) {
-                    seqPartials[k] = seqPartials[k - 1];
+
+            // TODO: investigate performance improvement using BST
+            insertIntoList(seqPartials, indexFirstGTE, partialLengthEntry);
+        }
+
+        const { unsequencedRecords } = combinedPartialLengths;
+        if (unsequencedRecords && removeClientOverlap && segment.localRemovedSeq !== undefined) {
+            const localSeq = segment.localRemovedSeq;
+            const localPartialLengthEntry: LocalPartialSequenceLength = {
+                seq,
+                localSeq,
+                clientId,
+                len: 0,
+                seglen: segmentLen,
+            };
+            let localIndexFirstGTE = 0;
+            const { partialLengths, overlappingRemoves } = unsequencedRecords;
```

Comment:
i think the below code would be clearer if we don't expand this, and just use `unsequencedRecords.`